dvtemplates/new_cdi_image.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+error_message() {
+    OS=$2
+    if [ $1 -eq 1 ]; then
+        echo -e "Unable to access data from the OS release URL"
+    elif [ $1 -eq 2 ]; then
+        echo -e "Invalid Image Version"
+    elif [ $1 -eq 3 ]; then
+        echo -e "Unable to connect to the OS Image repository"
+    elif [ $1 -eq 4 ]; then
+        echo -e "Container Image for the Latest $OS version is already present. Exiting"
+	exit 0
+    elif [ $1 -eq 5 ]; then
+        echo -e "Unable to push the image to the local Registry."
+    elif [ $1 -eq 6 ]; then
+        echo -e "Unable to push the image to the Quay Registry."
+    elif [ $1 -eq 7 ]; then
+        echo -e "End to End Tests fail."
+    elif [ $1 -eq 8 ]; then
+        echo -e "No OS Image is set. Exiting!"
+    fi
+    exit $1
+}
+
+trap 'error_message $?' EXIT
+
+#Insecure registry port
+port=5000
+
+# Check if TARGET_OS is set
+if [ -z "$TARGET_OS" ]; then
+    error_message 8
+fi
+
+FEDORA_OS="fedora"
+CENTOS="centos"
+
+# Set the Image registry and identify the latest version of the OS
+if [[ "${TARGET_OS}" == "${CENTOS}" ]]; then
+    cd ${PWD}/centos
+    OS_REPO="quay.io/kubevirt/centos-container-disk-images"
+    BASE_URL=https://cloud.centos.org/centos/
+    wget -qO index.html $BASE_URL || exit 1
+    OS_VERSION=`cat index.html | sed -e 's/.*>\([0-9]\+\)\/<.*/\1/' | sort -rn | head -n 1 | tr -d ' '`
+elif [[ "${TARGET_OS}" == "${FEDORA_OS}" ]]; then
+    cd ${PWD}/fedora
+    OS_REPO="quay.io/kubevirt/fedora-images"
+    RELEASE_URL=https://getfedora.org/releases.json
+    wget -qO release.html $RELEASE_URL || exit 1
+    OS_VERSION=`cat release.html | jq '.[] | select(.version|test("^[0-9]+$")) | .version' | sort -rn | uniq | head -n 1 | tr -d '"'`
+fi
+re='^[0-9]+$'
+if ! [[ $OS_VERSION =~ $re ]] ; then
+    error_message 2
+fi
+echo "Latest ${TARGET_OS} version is : ${OS_VERSION}"
+
+# Fetch the old image
+docker pull -a $OS_REPO
+OS_OLD_VERSION=$(docker images $OS_REPO --format "{{json .}}" | jq 'select(.Tag|test("^[0-9]+$")) | .Tag' | sort -rn | head -n 1 | tr -d '"')
+echo "${TARGET_OS} version in the Image Registry is : ${OS_OLD_VERSION}"
+
+if [ -z "${OS_OLD_VERSION}" ]; then
+    echo -e "No Container Image found in the registry"
+fi
+
+if [[ "$OS_VERSION" == "$OS_OLD_VERSION" ]]; then
+    error_message 4 "$TARGET_OS"
+fi
+
+#Download the latest OS Image
+if [[ "${TARGET_OS}" == "${CENTOS}" ]]; then
+    IMAGE_URL=${BASE_URL}${OS_VERSION}/x86_64/images/
+    wget -qO image.html $IMAGE_URL/CHECKSUM || exit 3
+    CentOS_IMAGE=$(grep ^#.*GenericCloud.*qcow2 image.html | sed -e 's/.*\(CentOS.*qcow2\).*/\1/')
```

Comment:
Please enclose the regex input to grep within quotes so it will not be interpreted by bash.

Reply:
Done